### DISCUSSION TITLE
RR for migrating legacy php app to go

### DISCUSSION THREAD

**Discussion by sunzo-digital** [roadrunner-server/roadrunner#1379] [Q&A]
Hi! My goal is to rewrite a legacy php 7.2 application based on the phalcon 3.4 framework with go. I'd like to do it in batches: write new features with go and then rewrite old code in go as well. Is RR suitable for this? In what ways can I do this?

**Comment by butschster** [comment]
Hi @sunzo-digital !
Could you explain the reason of rewriting from PHP to go?

**Reply by sunzo-digital** [reply]
I'm going to wtrite a lot of code for this project, but i don't want to deal with the legacy framework

**Reply by rustatian** [reply]
You may use our framework https://github.com/spiral/framework which supports all RR features, and start rewriting your legacy piece by piece. For example, you may redirect /v1 (old) calls to the old backend and /v2 (new) to the RR and your new worker. That can be achieved with any webserver you choose, or you may write a simple HTTP middleware in Go for the RR: https://youtu.be/f5fUSYaDKxo. 

But anyway, to work with RR, you have to write a worker. It's framework agnostic: https://roadrunner.dev/docs/php-worker/2.x/en.
